config/settings/base.py

```diff
@@ -85,7 +84,7 @@
 
 DATABASES = {
     "default": dj_database_url.config(
-        default="sqlite:///%s" % (join(BASE_DIR, "db.sqlite3"),)
+        default="sqlite:///{}".format(join(BASE_DIR, "db.sqlite3"))
```

Comment:
@fritzbrand (question for my own learning) we've moved away from `.format()` in a bunch of places, why is it the better choice in this case?

Reply:
TL;DR: Because `ruff` didn't tell us to use an f-string instead.

`ruff`'s rule for this is https://docs.astral.sh/ruff/rules/f-string/ which doesn't really explain much, but does link to the original source of the rules: https://pypi.org/project/pyupgrade/

Buried way down that page is the explanation: "note: `pyupgrade` is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are sufficiently complicated (as this can decrease readability)."

I'm guessing that in this case the `join(...)` is considered "sufficiently complicated", and since it doesn't seem that switching to an f-string will be any more readable than the `.format(...)` version, it probably makes sense to keep it as-is.

Thanks for asking! I certainly learned a bunch of stuff finding the answer. 😁